cobalt/media/base/sbplayer_pipeline.cc

```diff
@@ -1135,6 +1148,19 @@ void SbPlayerPipeline::OnNeedData(DemuxerStream::Type type,
                               type, max_buffers));
 }
 
+int SbPlayerPipeline::GetEstimatedMaxBuffers(TimeDelta write_duration,
+                                             TimeDelta time_ahead_of_playback) {
+  int estimated_max_buffers = 1;
+  if (!last_audio_sample_interval_.is_zero()) {
```

Comment:
`last_audio_sample_interval_` will be 0 for the first two writes.  So multi-sample writing will kick in starting from the third write.

We actually already know that we only have aac, opus, ac3, eac3, and iamf, and:
1. aac is 1024 frames per DecoderBuffer, the frequency can be retrieved from AudioDecoderConfig (and affected by HEAAC).
2. Opus is always 960 frames per DecoderBuffer for our use cases, and frequency can be retrieved the same way (and should be 48000).
3. e/ac3 is probably 1536 frames per DecoderBuffer.
4. I am not sure about IAMF, but @osagie98 should be able to provide more details.

So we can use these information when `last_audio_sample_interval_` is 0 to speed up the initial writings.
We should add a DCHECK() for codecs not listed above and fallback to the algorithm you are using.

Also, we can write 1 sample per 125ms of `write_duration` (given aac at 8000hz) at the very beginning as a simple solution.

Reply:
Added the predefined maximum numbers of sample writes to aac, opus, and e/ac3. Put iamf as TODO as we need the information from sub-stream to determine its frames (which is not implemented yet).